mythril/laser/ethereum/transaction/transaction_models.py

```diff
@@ -115,10 +117,17 @@ def initial_global_state_from_environment(self, environment, active_function):
 
         sender = environment.sender
         receiver = environment.active_account.address
-        value = environment.callvalue
+        value = (
+            environment.callvalue
+            if isinstance(environment.callvalue, BitVec)
+            else symbol_factory.BitVecVal(environment.callvalue, 256)
+        )
 
-        global_state.world_state.balances[sender] -= value
+        global_state.mstate.constraints.append(
```

Comment:
And yeah, even here. we should constrain it if only the balance isn't concrete.

Reply:
This constraint is to prevent you from sending more eth than you have. :smile:

Reply:
Oh! Yeah. I looked into the evm tests(`suicide0.json`), looks like this constraint is failing there. It might be a mistake or else evm should handle it with some min() comparision. That should be looked into.

Reply:
Oh. Reading `suicide0.json`... It attempts to send more ether than the account has. :joy: